src/cudnn/conv/filter/mod.rs

```diff
@@ -0,0 +1,167 @@
+mod descriptor;
+
+pub use descriptor::*;
+
+use core::ffi::c_void;
+
+use alloc::rc::Rc;
+
+use crate::cudarc::CudaUniquePtr;
+use crate::cudnn::sys::*;
+use crate::prelude::*;
+
+/// A convolution filter. Cloning this filter only clones the
+/// pointer to the [FilterDescriptor] and [Tensor4DData] and thus increases the
+/// reference count.
+pub struct Filter<T, const C_OUT: usize, const C_IN: usize, const H: usize, const W: usize> {
+    descriptor: Rc<FilterDescriptor<T, C_OUT, C_IN, H, W>>,
+    data: Tensor4DData<T, C_OUT, C_IN, H, W>,
+}
```

Comment:
It's weird how they have a separate descriptor for filters, I wonder why they don't just use a Tensor descriptor?